.github/workflows/main.yml

```diff
@@ -55,23 +55,9 @@ jobs:
     runs-on: ${{ matrix.platform }}
     continue-on-error: ${{ matrix.python == '3.14' }}
     steps:
-      - uses: actions/checkout@v4
-      - name: Install build dependencies
-        # Install dependencies for building packages on pre-release Pythons
-        # jaraco/skeleton#161
-        if: matrix.python == '3.14' && matrix.platform == 'ubuntu-latest'
-        run: |
-          sudo apt update
-          sudo apt install -y libxml2-dev libxslt-dev
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python }}
-          allow-prereleases: true
-      - name: Install tox
-        run: python -m pip install tox
-      - name: Run
-        run: tox
+    - uses: actions/checkout@v4
+    - name: Common
+      uses: ./.github/actions/common
```

Comment:
The other degradation that occurs with composite workflows is all the inner steps get folded into this "Common" step.

Reply:
...which is why I prefer reusable workflows to actions. Actions are there to encapsulate a single visual step, while workflows would show all.